packages/gatsby-worker/src/__tests__/integration.ts

```diff
@@ -282,4 +290,33 @@ describe(`gatsby-worker`, () => {
       )
     })
   })
+
+  describe(`messaging`, () => {
```

Comment:
Can we also add test cases for these things:

- Different messaging versions
- An error is thrown somewhere in the process and thus the payload isn't complete
- Sending a message to a worker with a `workerId` that doesn't exist

Reply:
> Different messaging versions

If you refer to https://github.com/gatsbyjs/gatsby/pull/32159/files#diff-c8c5155791c3198e422efc8e9982d95d341edf284a823fcfda4e700988a1e3daR18
here are few notes (I should probably add comment in code why it's even there):

This variable is intended to indicate shape of messenger API which currentlly consist of:
```ts
  onMessage: (listener: (msg: MessagesFromParent) => void) => void
  sendMessage: (msg: MessagesFromChild) => void
```

So that IF we change it (we don't have to), consumers (specifically ones that don't depend on `gatsby-worker` directly) can do runtime checks to see what API shape to use. This is just future proofing for handling of structured logs which will have to happen inside `gatsby-cli` at some level at least which won't depend on `gatsby-worker` directly, but probably will be able to grab messanger instace of it via `gatsby/internals` ( https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/internal.ts ). If we would change API of messenger then type would look like:

```
export type IGatsbyWorkerMessenger<
  MessagesFromParent = unknown,
  MessagesFromChild = MessagesFromParent
>  = {
  onMessage: (listener: (msg: MessagesFromParent) => void) => void
  sendMessage: (msg: MessagesFromChild) => void
  messagingVersion: 1
} | {
  completelyNewOnMessage: (listener: (msg: MessagesFromParent) => void) => void
  completelyNewSendMessage: (msg: MessagesFromChild) => void
  messagingVersion: 2
}
```

so that consumers pretty much have to check `messagingVersion` before interacting to get type safe support for current and previous API shapes

That said - there is no other version of messaging available so most I could do to test it is do some TS type testing for now (just not sure how)

> An error is thrown somewhere in the process and thus the payload isn't complete

I don't understand what you mean with this / what scenario are you thinking about. If the error would happen unrelated to messages, then it's up to whatever make use of messaging API to handle those errors? Can you provide some example of what you thinking here to check? 

> Sending a message to a worker with a workerId that doesn't exist

Yup, that sounds like sensible thing to test 👍

Reply:
Okay, `messagingVersion` seems fine this way then, no need to add a test :)

Regarding the second bulletpoint:

What I meant is that if we can test this behavior:

```js
let getWasPonged = function (): boolean {
  throw new Error(`gatsby-worker messenger not available`)
}
```

So if for some reason the messenger is not available we at least get an error

Reply:
Mhh, maybe scratch that. This is more testing the test setup than our actual code.

Reply:
> Sending a message to a worker with a workerId that doesn't exist

Added in https://github.com/gatsbyjs/gatsby/pull/32159/commits/deea3dcde910922739b2227985e07e8c90fef0ce